### ISSUE TITLE
evdi crashes DisplayLinkManager when moving mouse to connected display

### ISSUE THREAD

**Issue by winterwurzel** [DisplayLink/evdi#446]
* Are you using the latest driver?
    using the latest fedora 39 release from: https://github.com/displaylink-rpm/displaylink-rpm/releases/download/v5.8.0-1/fedora-39-displaylink-1.14.1-2.x86_64.rpm
  * Are you using the latest EVDI version?
    Yes, evdi-1.14.1
  * If you are using a DisplayLink device, have you checked 'troubleshooting'
    on DisplayLink's website?
    Yes
  * Is this issue related to evdi/kernel?
    Yes, from what I understand in the stacktrace

  * Linux distribution and its version: `Fedora 39`
  * Linux kernel version: `Linux V 6.6.8-200.fsync.fc39.x86_64 #1 SMP PREEMPT_DYNAMIC TKG Fri Dec 29 23:43:56 UTC 2023 x86_64 GNU/Linux`
  * Xorg version (if used): `Wayland`
  * Desktop environment in use: `KDE Plasma Version 5.27.10`

## Description
As soon as I move the mouse cursor to a monitor connected via displaylink it crashes. Moving any type and number of windows on the screen (via keyboard shortcuts) works flawlessly.

Syslog stacktrace can be found further below. 

## Logs
  * syslog stacktrace: https://pastebin.com/4m2WjkgU
  * `/var/lib/dkms/evdi/kernel-[version]/log/make.log`: https://pastebin.com/4cqRU3C1
  * `/var/log/displaylink/displaylink.log`:
```
Creating symlink /var/lib/dkms/evdi/1.14.1/source -> /usr/src/evdi-1.14.1
Sign command: /lib/modules/6.6.8-200.fsync.fc39.x86_64/build/scripts/sign-file
Signing key: /var/lib/dkms/mok.key
Public certificate (MOK): /var/lib/dkms/mok.pub

Building module:
Cleaning build area...
make -j24 KERNELRELEASE=6.6.8-200.fsync.fc39.x86_64 all INCLUDEDIR=/lib/modules/6.6.8-200.fsync.fc39.x86_64/build/include KVERSION=6.6.8-200.fsync.fc39.x86_64 DKMS_BUILD=1...
Signing module /var/lib/dkms/evdi/1.14.1/build/evdi.ko
Cleaning build area...

evdi.ko.xz:
Running module version sanity check.
 - Original module
   - No original module exists within this kernel
 - Installation
   - Installing to /lib/modules/6.6.8-200.fsync.fc39.x86_64/extra/
depmod....
```

**Comment by MariusLer** [comment]
I have the same issue.

Linux distribution and its version: Ubuntu 23.10
Linux kernel version: Linux 6.5.0-14-generic #14-Ubuntu SMP PREEMPT_DYNAMIC Tue Nov 14 14:59:49 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
Xorg version (if used): Wayland
Desktop environment in use: KDE Plasma Version 5.110.0
Pc: Lenovo thinkpad T14 Gen3 (AMD)
Docking Station: Lenovo ThinkPad Hybrid ISB_C with USB-A Dock (DUD9011D1)

After a lot of crashes from driver, the issue will suddenly disappear and moving the cursor works just fine.
The issue comes back after reconnecting or restarting my laptop.

**Comment by qdel** [comment]
Hello,

New user of a display link dock, I face the exact same issue.

Screens are working fine (Moving glxgears using keyboard on all of them).

Did you find a workaround?

**Comment by mnixry** [comment]
After conducting a debug, I've determined that the problem might actually be a bug associated with DisplayLink, not evdi. The stack trace shows a failure in the `cursor_set_handler` function, which is passed from DisplayLink. However, since DisplayLink is a proprietary, fully stripped program, we have limited options for fixing this issue.

Here's the relevant part of the stack trace:

```
DisplayLinkManager v5.8.0 (6.3.33.0)
```

```
#0  0x00007f0fca175587 n/a (libc.so.6 + 0x15b587)
#1  0x00007f0fca49101a evdi_handle_events (libevdi.so + 0x401a)
#2  0x000055c813588bee n/a (DisplayLinkManager + 0x244bee)
#3  0x000055c813465fc4 n/a (DisplayLinkManager + 0x121fc4)
#4  0x000055c8134e3c60 n/a (DisplayLinkManager + 0x19fc60)
#5  0x00007f0fca0a955a n/a (libc.so.6 + 0x8f55a)
#6  0x00007f0fca126a3c n/a (libc.so.6 + 0x10ca3c)
```

The issue appears to stem from this part of the code:

https://github.com/DisplayLink/evdi/blob/9905b9bfed8b7e49befa0530bfb6e464380d938f/library/evdi_lib.c#L949-L961

---

If you're facing this issue, there's a workaround to prevent the program from crashing:

```diff
 case DRM_EVDI_EVENT_CURSOR_SET: 
- 	if (evtctx->cursor_set_handler) { 
+	if (evtctx->cursor_set_handler && false) { // disable cursor set handler
 	 	struct drm_evdi_event_cursor_set *event = 
 			(struct drm_evdi_event_cursor_set *) e; 
 		struct evdi_cursor_set cursor_set = to_evdi_cursor_set(handle, event); 
  
 		if (cursor_set.enabled && cursor_set.buffer == NULL) 
 			evdi_log("Error: Cursor buffer is null!"); 
 		else 
 			evtctx->cursor_set_handler(cursor_set, 
 						   evtctx->user_data); 
 	} 
```

This workaround will prevent the mouse cursor from appearing on the display, but it will stop the crashes.

**Comment by dclasen-qd** [comment]
Same issue over here using a DELL D6000 Dock. The workaround, setting the env variable `KWIN_DRM_NO_AMS=1`, mentioned in https://github.com/DisplayLink/evdi/issues/459#issuecomment-1980882934 fixed the crashes of `displaylink.service`.

**Comment by displaylink-emajewsk** [comment]
Fixed in 1.14.3
